server/controllers/lettersControllers/lettersControllers.js

```diff
@@ -40,8 +40,9 @@ exports.getAllLetters = async (req, res) => {
 
 exports.getLettersById = async (req, res) => {
     try {
-        const {letterId} = req.params.id
```

Comment:
We were desctructuring something that wasn't an object. So the value `letterId` was undefined when looking it up in the collection. Letters would be null.